design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,656 @@
+and `KubernetesCluster` claim the two would have connectivity:
+
+```yaml
+---
+# A Network managed resource.
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Network
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  providerRef:
+    namespace: crossplane-system
+    name: example
+  nameFormat: mycoolnetwork
+  autoCreateSubnetworks: false
+---
+# A Subnetwork managed resource.
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Subnetwork
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  providerRef:
+    namespace: crossplane-system
+    name: example
+  nameFormat: mycoolsubnetwork
+  # Create this subnet in the Network we created previously.
+  # mycoolproject must match the Crossplane GCP Provider project.
+  # mycoolnetwork must match the above Network managed resource's name.
+  network: projects/mycoolproject/global/networks/mycoolnetwork
+  ipCidrRange: 172.16.10.0/24
+  region: us-central1
+  privateIpGoogleAccess: true
+  secondaryIpRanges:
+  - rangeName: pods
+    ipCidrRange: 10.0.0.0/8
+  - rangeName: services
+    ipCidrRange: 172.16.20.0/24
+---
+# A ResourceClass that sastisfies MySQLInstance claims using CloudSQLInstance
+# managed resources.
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceClass
+metadata:
+  namespace: crossplane-system
+  name: default-mysqlinstance
+  labels:
+    # This resource class is the default for MySQLInstance resource claims.
+    mysqlinstance.database.crossplane.io/default: "true"
+providerRef:
+  namespace: crossplane-system
+  name: example
+parameters:
+  nameFormat: mycoolname
+  databaseVersion: MYSQL_5_6
+  region: us-central1
+  tier: db-n1-standard-1
+  dataDiskSizeGb: "50"
+  dataDiskType: PD_SSD
+  # Allow access to this CloudSQL instance from the Network we created previously.
+  # mycoolproject must match the Crossplane GCP Provider project.
+  # mycoolnetwork must match the above Network managed resource's name.
+  privateNetwork: /projects/mycoolproject/global/networks/mycoolnetwork
+---
+# A ResourceClass that sastisfies KubernetesCluster claims using GKECluster
+# managed resources.
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceClass
+metadata:
+  namespace: crossplane-system
+  name: default-kubernetescluster
+  labels:
+    # This resource class is the default for KubernetesCluster resource claims.
+    kubernetescluster.compute.crossplane.io/default: "true"
+providerRef:
+  namespace: crossplane-system
+  name: example
+parameters:
+  clusterVersion: "1.12"
+  machineType: n1-standard-2
+  numNodes: "3"
+  zone: us-central1-a
+  # Create nodes in the mycoolsubnetwork subnetwork of the mycoolnetwork network.
+  # mycoolnetwork must match the above Network managed resource's name.
+  # mycoolsubnetwork must match the above Subnetwork managed resource's name.
+  network: mycoolnetwork
+  subnetwork: mycoolnetwork
+  # Enable VPC native subnetworks.
+  enableIPAlias: "true"
```

Comment:
Are the double quotes needed?

Reply:
Sadly yes, because under the hood `parameters` is a `map[string]string`. This is one of the shortcomings strongly typed resource classes hopes to address.

Reply:
kk. Could we add a comment to the field so that future readers have the same context?